app/core/model/transfer/repo.go

```diff
@@ -0,0 +1,20 @@
+package transfer
+
+import (
+	"context"
+	"stonehenge/app/core/types/id"
+)
+
+//go:generate moq -fmt goimports -out repo_mock.go . Repository:RepositoryMock
+
+// Repository is the data access layer for the transfer entity
+type Repository interface {
+	// List gets all transfers existing
+	List(ctx context.Context, filter Filter) ([]Transfer, error)
+
+	// Get gets the transfer with the ID specified
+	Get(ctx context.Context, id id.ID) (*Transfer, error)
```

Comment:
tem algum motivo em específico para retornar um ponteiro de `Transfer`?

Reply:
Até onde aprendi em Go, só posso retornar um valor nulo se ele for error ou ponteiros, não? 
Sob esse ponto de vista, todas as funções que montei que podem retornar erro também, coloquei para retornar o ponteiro do objeto principal, já que, se houver erro, ele vai retornar nulo. Está assim também na interface do repositório de Accounts na função GetBalance, se não me engano. O que você acha disso? Vou marcar o @RegisMelgaco pois ele também pontuou essa questão de ponteiros

Reply:
> Até onde aprendi em Go, só posso retornar um valor nulo se ele for error ou ponteiros, não?

É isso mesmo!

> Sob esse ponto de vista, todas as funções que montei que podem retornar erro também, coloquei para retornar o ponteiro do objeto principal, já que, se houver erro, ele vai retornar nulo.

Eu entendo a motivação, mas acredito que isso não seja necessário em Go uma vez que trabalhamos com múltiplos retornos de função, já que não utilizamos o objeto principal para verificar se houve erro, o erro vem separado. Adicionalmente, costumamos retornar o objeto principal como a struct vazia para evitar confusão de receber dois retornos válidos.

Adicionar o ponteiro para tratar apenas o "null" pode levar a problemas de nullpointer e adiciona complexidade de que essa estrutura pode estar sendo alterada em um local indevido. 

Eu costumo usar ponteiro apenas quando é algo que requer mesmo uma única estrutura que compartilha os recursos com outras estruturas e funções, caso contrário eu opto por usar a clássica cópia :)